packages/terra-application-navigation/src/utility-menu/_UtilityMenu.jsx

```diff
@@ -72,6 +72,7 @@ const UtilityMenu = ({
       key: utilityMenuSettingsKey,
       text: intl.formatMessage({ id: 'Terra.applicationNavigation.utilityMenu.settings' }),
       icon: <IconSettings />,
+      dataAttrs: { 'data-navigation-utility-item-settings': 'true' },
```

Comment:
Are dataAttrs a part of the previous API? Could the attributes be spread through custom props without the new prop? If dataAttrs stays as a first class props, which component should have documentation for it in the prop types?

Reply:
@StephenEsser 
> Are dataAttrs a part of the previous API?

No.

Please correct me if I am wrong, I parse and pass down `dataAttrs` as a custom prop [here](https://github.com/cerner/terra-framework/pull/1018/files#diff-1c8abd4cb6c3f757361be6e3f68c4646R110). Do you mean to say this is a first class prop at this [point](https://github.com/cerner/terra-framework/pull/1018/files#diff-d8360db5f441f93b4f8126e278dc25caR201).

Reply:
It becomes part of the API contract between the component that sets the `dataAttrs` and the component that reads it. These appear to be a private implementations using objects so it may not be too big of a deal if it's not exposed to consumers.

Reply:
@lokesh-0813 I'd at least add `dataAttrs` to the shape defined here for the menuItems: https://github.com/cerner/terra-framework/pull/1018/files#diff-d8360db5f441f93b4f8126e278dc25caR49

I don't think the line Stephen called out will be a problem (unless `item` itself is undefined). But please double check that execution without a dataAttrs prop for an item works appropriately, we don't want it to be a required property.

Reply:
updated here a6f2c8a982409414faaefd171e7462ffb03dcd17